test/src/unittests/tonal/test_audio2midi.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2024  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+from numpy import mean, array, float32, square
+from pathlib import Path
+
+
+class TestAudio2Midi(TestCase):
+    def testEmpty(self):
+        self.assertComputeFails(Audio2Midi(), [])
+
+    def testZero(self):
+        pitch, loudness, message_type, midi_note, time_compensation = Audio2Midi()(
+            zeros(32)
+        )
+        self.assertEqual(pitch, 0)
+        self.assertEqual(loudness, 0.0)
+        self.assertEqual(message_type, [])
+        self.assertEqual(midi_note.tolist(), array([], dtype=float32).tolist())
+        self.assertEqual(time_compensation.tolist(), array([], dtype=float32).tolist())
+
+    def assessNoteList(
+        self,
+        reference_path: str,
+        estimated: list,
+        n_notes_tolerance: int = 0,
+        onset_tolerance: float = 0.01,
+        offset_tolerance: float = 0.01,
+        midi_note_tolerance: int = 0,
+    ):
+        # read the expected notes file manually annotated
+        expected_notes = numpy.load(join(filedir(), reference_path))
+        print("Expected notes:")
+        print(expected_notes)
+
+        print("\ndiffs")
+        print(array(estimated) - expected_notes[:, 1:])
+
+        # estimate the number of notes for expected and detected
+        n_detected_notes = len(estimated)
+        n_expected_notes = len(expected_notes)
+
+        # estimate the onset error for each note and estimate the mean
+        onset_mse = mean(
```

Comment:
This check the mean error, however, are there any limits on individual error we should check?

Reply:
Off course, we can add a local assessment. I was checking other tests and they used a similar strategy.